File: Web/Parking.UI/Content/Scripts/App/Models/CheckinNotification.js
/**
* Checkin Notification
*
* @license Copyright 2012. The JSONS
*/
namespace("Parking.App.Base");
namespace("Parking.App.Models");

(function ($, parking, undefined) {
    var config         = parking["Configuration"];
    var appbase        = parking["App"]["Base"];
    var appmodels      = parking["App"]["Models"]; 

    /**
     *
     * @extends Parking.App.Base.Model
     */
    appmodels.CheckinNotification = appbase.Model.extend({
        
        /**
         * Primary Key
         * @type {string}
         */
        "idAttribute": "NotificationId",

        /**
         * @enum {Object}
         */
        "defaults": {
            "NotificationId": "",
            "CheckInId": 0,
            "UserId": 0,
            "SpaceId": 0,
            "StartTime": new Date(),
            "EndTime": null, 
            "RegisteredFrom": 0,
            "RegisteredBy": 0,
            "LastModified": new Date(),
            "NotificationType": 0,
            "NotificationDesc": ""
        },

        /**
         * @constructor
         */
        "initialize": function() {
        }

    });

})(jQuery, Parking);

File: API/Parking.API.Models/Views/CheckinNotification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace Sieena.Parking.API.Models.Views
{
    public class CheckinNotification
    {
        public CheckinNotification(CheckinNotification c, Checkin.NotificationType type)
        {
            CheckInId = c.CheckInId;
            SpaceId = c.SpaceId;
            UserId = c.UserId;
            StartTime = c.StartTime;
            EndTime = c.EndTime;
            LastModified = (type == Checkin.NotificationType.Checkout && c.EndTime.HasValue) ? c.EndTime.Value : c.StartTime;
            RegisteredBy = c.RegisteredBy;
            RegisteredFrom = c.RegisteredFrom;
            NotificationType = (int)((type == Checkin.NotificationType.Checkout && c.EndTime.HasValue) ? Checkin.NotificationType.Checkout : Checkin.NotificationType.Checkin);
        }

        public CheckinNotification(Checkin c, Checkin.NotificationType type)
        {
            CheckInId = c.CheckInId;
            SpaceId = c.SpaceId;
            UserId = c.UserId;
            StartTime = c.StartTime;
            EndTime = c.EndTime;
            LastModified = (type == Checkin.NotificationType.Checkout && c.EndTime.HasValue) ? c.EndTime.Value : c.StartTime;
            RegisteredBy = c.RegisteredBy;
            RegisteredFrom = c.RegisteredFrom;
            NotificationType = (int)((type == Checkin.NotificationType.Checkout && c.EndTime.HasValue) ? Checkin.NotificationType.Checkout : Checkin.NotificationType.Checkin);
        
        }

        [Display(Name = "Notification Id")]
        public string NotificationId { 
            get {
                return string.Format("{0}{1}", this.NotificationDesc.ToUpper(), this.CheckInId);
            } 
        }

        [Display(Name = "Checkin Id")]
        public int CheckInId { get; set; }

        [Display(Name = "User Id")]
        public int UserId { get; set; }

        [Display(Name = "Space Id")]
        public int SpaceId { get; set; }

        [Display(Name = "Start Time")]
        public DateTime StartTime { get; set; }

        [Display(Name = "End Time")]
        public DateTime? EndTime { get; set; }

        [Display(Name = "Notification Type")]
        public int NotificationType { get; set; }

        [Display(Name = "Notification Description")]
        public string NotificationDesc { get { return ((NotificationType == (int)Checkin.NotificationType.Checkin) ? "in" : "out"); } }

        [Display(Name = "Last Modified")]
        public DateTime LastModified { get; set; }

        [Display(Name = "Registered From")]
        public int RegisteredFrom { get; set; }

        [Display(Name = "Registered By")]
        public int RegisteredBy { get; set; }
         
    }
}
